internal/parser/simple_parser_rules.go

```diff
@@ -918,2436 +929,5036 @@ func (p *simpleParser) parseConflictClause(r reporter) (clause *ast.ConflictClau
+	expr.FunctionName = functionName
+	next, ok := p.lookahead(r)
 	if !ok {
 		return
 	}
-	if next.Type() == token.KeywordIf {
-		stmt.If = next
+	if next.Type() == token.Delimiter && next.Value() == "(" {
+		expr.LeftParen = next
 		p.consumeToken()
+
 		next, ok = p.lookahead(r)
 		if !ok {
 			return
 		}
-		if next.Type() == token.KeywordNot {
-			stmt.Not = next
+		if next.Type() == token.KeywordDistinct {
+			expr.Distinct = next
+			p.consumeToken()
+			expr.Expr = p.parseExprSequence(r)
+		} else if next.Type() == token.BinaryOperator {
+			expr.Asterisk = next
+			p.consumeToken()
+		} else if next.Type() == token.Delimiter && next.Value() == ")" {
+			expr.RightParen = next
 			p.consumeToken()
-			next, ok = p.lookahead(r)
-			if !ok {
-				return
-			}
-			if next.Type() == token.KeywordExists {
-				stmt.Exists = next
-				p.consumeToken()
-			} else {
-				r.unexpectedToken(token.KeywordExists)
-			}
 		} else {
-			r.unexpectedToken(token.KeywordNot)
+			expr.Expr = p.parseExprSequence(r)
 		}
-	}
 
-	schemaNameOrIndexName, ok := p.lookahead(r)
-	if !ok {
-		return
-	}
-	if schemaNameOrIndexName.Type() == token.Literal {
-		// This is the case where there might not be a schemaName
-		// We assume that its the table name in the beginning and assign it
-		stmt.IndexName = schemaNameOrIndexName
-		p.consumeToken()
-		next, ok = p.lookahead(r)
-		if !ok {
+		next, ok = p.optionalLookahead(r)
+		if !ok || next.Type() == token.EOF || next.Type() == token.StatementSeparator {
 			return
 		}
-		// If we find the signs of there being a SchemaName,
-		// we over-write the previous value
-		if next.Value() == "." {
-			stmt.SchemaName = schemaNameOrIndexName
-			stmt.Period = next
-			p.consumeToken()
-			indexName, ok := p.lookahead(r)
-			if !ok {
-				return
-			}
-			stmt.IndexName = indexName
+		if next.Type() == token.Delimiter && next.Value() == ")" {
+			expr.RightParen = next
 			p.consumeToken()
 		}
-	}
 
-	next, ok = p.lookahead(r)
-	if !ok {
-		return
-	}
-	if next.Type() == token.KeywordOn {
-		stmt.On = next
-		p.consumeToken()
-	}
+		next, ok = p.optionalLookahead(r)
+		if !ok || next.Type() == token.EOF || next.Type() == token.StatementSeparator {
+			return
+		}
+		if next.Type() == token.KeywordFilter {
+			expr.FilterClause = p.parseFilterClause(r)
+		}
 
-	next, ok = p.lookahead(r)
-	if !ok {
-		return
+		next, ok = p.optionalLookahead(r)
+		if !ok || next.Type() == token.EOF || next.Type() == token.StatementSeparator {
+			return
+		}
+		if next.Type() == token.KeywordOver {
+			expr.OverClause = p.parseOverClause(r)
+		}
+
+		next, ok := p.optionalLookahead(r)
+		if !ok || next.Type() == token.EOF || next.Type() == token.StatementSeparator {
+			return
+		}
+		returnExpr := p.parseExprRecursive(expr, r)
+		if returnExpr != nil {
+			expr = returnExpr
+		}
 	}
-	if next.Type() == token.Literal {
-		stmt.TableName = next
-		p.consumeToken()
+	return
+}
+
+// parseExprSequence parses a sequence of exprs separated by ","
```

Comment:
Full sentences please.
```suggestion
// parseExprSequence parses a sequence of exprs separated by ",".
```